src/overlays/actors/ovl_En_Toryo/z_en_toryo.c

```diff
@@ -29,27 +30,343 @@ const ActorInit En_Toryo_InitVars = {
+}
+
+s32 func_80B203D8(EnToryo* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = PLAYER;
+    s32 ret = 1;
+
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+        case 5:
+            ret = 1;
+            break;
+        case 4:
+            if (func_80106BC8(globalCtx)) {
+                if (globalCtx->msgCtx.choiceIndex == 0) {
+                    func_80106CCC(globalCtx);
+                    this->actor.parent = NULL;
+                    player->exchangeItemId = EXCH_ITEM_NONE;
+                    globalCtx->msgCtx.msgMode = 0x37;
+                    this->actor.textId = 0x601B;
+                    ret = 3;
+                } else {
+                    this->actor.textId = 0x606F;
+                    ret = 2;
+                }
+            }
+            break;
+        case 6:
+            switch (this->actor.textId) {
+                case 0x5028:
+                    ret = 1;
+                    if (func_80106BC8(globalCtx) != 0) {
+                        gSaveContext.infTable[23] |= 0x4;
+                        ret = 0;
+                    }
+                    break;
+                case 0x601B:
+                    ret = 1;
+                    if (func_80106BC8(globalCtx) != 0) {
+                        ret = 4;
+                    }
+                    break;
+                case 0x606F:
+                    ret = 1;
+                    if (func_80106BC8(globalCtx) != 0) {
+                        gSaveContext.infTable[23] |= 0x2;
+                        ret = 0;
+                    }
+                    break;
+                case 0x606A:
+                    ret = 1;
+                    if (func_80106BC8(globalCtx) != 0) {
+                        gSaveContext.infTable[23] |= 0x1;
+                        ret = 0;
+                    }
+                    break;
+                case 0x606B:
+                case 0x606C:
+                case 0x606D:
+                case 0x606E:
+                default:
+                    ret = 1;
+                    if (func_80106BC8(globalCtx) != 0) {
+                        ret = 0;
+                    }
+            }
+    }
+    return ret;
+}
+
+s32 func_80B205CC(EnToryo* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = PLAYER;
+    s32 ret = 5;
+
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+        case 5:
+            ret = 5;
+            break;
+        case 6:
+            if (func_80106BC8(globalCtx) != 0) {
+                ret = 0;
+            }
+    }
+    return ret;
+}
+
+u32 func_80B20634(EnToryo* this, GlobalContext* globalCtx) {
+    u32 ret;
+
+    if (this->unk_1E0 != 0) {
+        if (this->unk_1E0 == 10) {
+            func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+            if ((gSaveContext.infTable[23] & 0x2)) {
```

Comment:
Relevant in the rest of the file but imo there is no need for double parens in single conditions like this
```suggestion
            if (gSaveContext.infTable[23] & 0x2) {
```